### PR TITLE
#32 - Added in a new media type of IFrame.

### PR DESCRIPTION
**Pull request by ageddesi**
I am sorry about the formatting, I could not tell if it was my editor or the settings in the project.

Also I just want to clarify that I have added a pipe that sanitises the URL. we should somehow make it clear to the user they can only embed sites that have given cross cors permissions.

### CODE REVIEW

File: src/app/core/pipes/safe-url/safe-url.pipe.spec.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { SafePipe } from "./safe-url.pipe";
```
[the remark this answers is not in the document]
Comment (ageddesi) [reply src/app/core/pipes/safe-url/safe-url.pipe.spec.ts]: Removed

File: src/app/core/pipes/pipes-module.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { NgModule } from "@angular/core";
```
[the remark this answers is not in the document]
Comment (ageddesi) [reply src/app/core/pipes/pipes-module.ts]: Renamed

File: src/app/manage/media/media-overview/media-info/media-info.component.html [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
+    class="preview"
+  >
+    <img
+      [src]="domSanitizer.bypassSecurityTrustUrl(info.previewUrl || info.path)"
+      alt="{{ info.name }}"
+      class="preview-img"
+    />
+  </div>
+  <div *ngIf="info.type === 3 && (info.previewUrl || info.path)">
+    <iframe [src]="info.path || info.previewUrl | safeurl"></iframe>
```
[the remark this answers is not in the document]
Comment (ageddesi) [reply src/app/manage/media/media-overview/media-info/media-info.component.html]: Updated

File: src/app/shared/components/dialogs/media-edit/media-edit.component.html [hunk 4]
```diff
@@ -64,47 +92,63 @@ <h2 mat-dialog-title> {{ data?.id ? 'Update Media: "' + form.value.name + '"' :
...
+        />
         <span matSuffix>ms</span>
 
         <mat-error *ngIf="form.controls['playLength'].hasError('required')">
           How long should it be visible on screen?
         </mat-error>
       </mat-form-field>
     </div>
-    <ng-container *ngIf="form.value.type !== 0">
+    <ng-container *ngIf="form.value.type !== 0 || form.value.type === 3">
```
[the remark this answers is not in the document]
Comment (ageddesi) [reply src/app/shared/components/dialogs/media-edit/media-edit.component.html]: Made it so it wont show if an IFrame for now, we can always bring back if needed

**Comment by ageddesi** [comment]
I need to make a change based on the merge from master into my branch, will let you know when this is done.

**Comment by ageddesi** [comment]
After the merge this is what it looks like for me, is this correct?
image

**Comment by negue** [comment]
Yes those are the colors that will be applied - @whitep4nth3r just left it there :P   - I think I can try to apply those colors today. (and then this panel will be gone as well )

Still need to test this, but its only missing one part, the changes in `target-screen.component.html` are missing (where the iframe would be visible )

If you want I can do this part after the merge :)

**Comment by whitep4nth3r** [comment]
I might do some work on this tonight - the colours still aren't final - I
need to see them in context!

**Comment by negue** [comment]
I wanted to re-align some of these html files (to prevent other conflicts with @whitep4nth3r's PR)  but that didn't work out (I don't know how to commit on a fork :D ) 

Thank you for the first contribution!!! 👍  :) 

I think we're close to release the next UX-improved version soon and then your Iframe-type will be released as well :)

**Comment by ageddesi** [comment]
Thanks a lot, let me know if there is any issues and I will resolve them

**Comment by negue** [comment]
@all-contributors please add @ageddesi for code

**Pull request merged**